config/config.go

```diff
@@ -102,6 +112,9 @@ func (c *Config) Ensure(fromCloud bool, logger golog.Logger) error {
 			}
 			logger.Errorw("module config error; starting robot without module", "name", c.Modules[idx].Name, "error", err)
 		}
+		if err := c.validateUniqueResource(seenResources, c.Modules[idx].Name); err != nil {
+			return err
```

Comment:
I think @cheukt was suggesting something like
```
if c.DisablePartialStart {
    return err
}
logger.Error(err)
```

Reply:
oh so not error out but just log? Not sure I completely follow how that is different to what I had before

Reply:
`DisablePartialStart` is a field that can be specified on a robot config. If set to true, the robot should not continue construction if there is any issue in config validation. If set to false, the robot should simply log the error but continue construction. The code I just wrote above will return the error if we've disabled partial start and stop construction. If partial start is enabled, though, it just logs the error.

Reply:
`DisablePartialStart` is set to false by default, so the default behavior will be to just log the error and not stop construction.

Reply:
ah okay i see thank you! I will clean it up that way